kube/src/config/file_config.rs

```diff
@@ -38,7 +38,7 @@ pub struct Preferences {
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct NamedExtension {
     pub name: String,
-    pub extension: String,
+    pub extension: serde_yaml::Mapping,
```

Comment:
I would use `serde_json::Value` instead, because `serde_json` exposes richer API than `serde-yaml` (such as `serde_json::Value::pointer`).

Reply:
Done - thank you for the suggestion!